aioxmpp/muc/service.py

```diff
@@ -618,18 +633,28 @@ def _match_tracker(self, message):
         try:
             tracker = self._tracking_by_id[message.id_]
         except KeyError:
+            key = message.from_, _extract_one_pair(message.body)
```

Comment:
Yeah, but we have to remember that this is called for *every* inbound message, even if the MUC does not rewrite IDs (because inbound messages except those which are reflections (usually the minority) fail the check above). I would prefer to stick to O(1) checks here. Especially if the MUC is actively evil and actively breaks the tracking: In that case it could make linear checks arbitrarily expensive. This way, it can only break tracking.